src/components/Editor/Preview/Popup.js

```diff
@@ -96,9 +102,22 @@ export class Popup extends Component {
   }
 
   renderObjectPreview(expression: string, root: Object) {
-    return (
-      <div className="preview-popup">{this.renderObjectInspector(root)}</div>
-    );
+    let content = null;
+    const { loadedObjects } = this.props;
+    const getObjectProperties = id => loadedObjects[id];
+    const roots = this.getChildren(root, getObjectProperties);
+
+    if (!roots) {
+      return null;
+    }
+
+    if (isReactComponent(roots)) {
+      content = <ReactPopup roots={roots} />;
```

Comment:
At the moment, i want to avoid styling so we have some more flexibility. I think this means we can avoid the `ReactPopup` and just re-use the OI :)

Reply:
Lol kay. I actually did that initially, I just found it to be too ugly compared to your pretty mockup